### PR TITLE
[Testing] Add application staking with Makefile targets for app and gateways

### PR DESCRIPTION
**Pull request by red-0ne**
## Summary

* Add Makefile targets for `Application` and `Gateway` staking/unstaking e2e test
* Add `app_stake.feature` e2e test with hadcoded `anvil` service
* Rename `stake.feature` to `gateway_stake.feature`

## Issue

- [MVT features table](https://www.notion.so/buildwithgrove/Minimum-Viable-Testnet-Success-Criteria-baa2392d0c934f408a277c6afa1b226c?pvs=4#e9bed3470316427998c8ab604bbfdff6)

## Type of change

Select one or more:

- [ ] New feature, functionality or library
- [ ] Bug fix
- [x] Code health or cleanup
- [ ] Documentation
- [ ] Other (specify)

## Testing

**Documentation changes** (only if making doc changes)
- [ ] `make docusaurus_start`; only needed if you make doc changes

**Local Testing** (only if making code changes)
- [x] **Unit Tests**: `make go_develop_and_test`
- [x] **LocalNet E2E Tests**: `make test_e2e`
- See [quickstart guide](https://dev.poktroll.com/developer_guide/quickstart) for instructions

**PR Testing** (only if making code changes)
- [ ] **DevNet E2E Tests**: Add the `devnet-test-e2e` label to the PR.
    - **THIS IS VERY EXPENSIVE**, so only do it after all the reviews are complete.
    - Optionally run `make trigger_ci` if you want to re-trigger tests without any code changes
    - If tests fail, try re-running failed tests only using the GitHub UI as shown [here](https://github.com/pokt-network/poktroll/assets/1892194/607984e9-0615-4569-9452-4c730190c1d2)

## Sanity Checklist

- [x] I have tested my changes using the available tooling
- [ ] I have commented my code
- [x] I have performed a self-review of my own code; both comments & source code
- [ ] I create and reference any new tickets, if applicable
- [ ] I have left TODOs throughout the codebase, if applicable

### CODE REVIEW

File: e2e/tests/stake_app.feature [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+    Scenario: User can stake an Application
+        Given the user has the pocketd binary installed
+        And the "application" for account "app2" is not staked
+        And the account "app2" has a balance greater than "1000070" uPOKT
+        When the user stakes a "application" with "1000070" uPOKT from the account "app2"
+        Then the user should be able to see standard output containing "txhash:"
+        And the user should be able to see standard output containing "code: 0"
+        And the pocketd binary should exit without error
+        # TODO_TECHDEBT(@Olshansk, @red-0ne): Replace these time-based waits with event listening waits
```
Comment (Olshansk) [inline e2e/tests/stake_app.feature:11]:
```suggestion
        # TODO_TECHDEBT: Replace these time-based waits with listeners for on-chain Stake events once they are implemented
```

File: e2e/tests/init_test.go [hunk 2]
```diff
@@ -193,6 +193,10 @@ func (s *suite) TheUserStakesAWithUpoktFromTheAccount(actorType string, amount i
 	require.NoError(s, err, "error creating config file in %q", path.Join(os.TempDir(), configPathPattern))
 
 	configContent := fmt.Sprintf(`stake_amount: %d upokt`, amount)
+	if actorType == "application" {
```
Comment (Olshansk) [inline e2e/tests/init_test.go:196]: Thoughts on:

1. Adding `service` to the function name?
2. If the `actorType` == `gateway`, create an assertion that says "must be empty"
3. We can reuse this for suppliers & applications
 -> Will need to update some of the other tests

**Review by Olshansk — CHANGES_REQUESTED** [review]

File: e2e/tests/init_test.go (hunk 2 above)
Comment (red-0ne) [reply e2e/tests/init_test.go:196]: 1. Adding `service` to the function name will make it sound unnatural for the `gateway` case. I prefer either:
  * Have another function for `gateway`s that wraps this one and omit the `service` part
  * Just add the service variable before the actor type which sounds:
    * `the user stakes a "svc" "application" with "1000070" uPOKT from the account "app2"
    * `the user stakes a "gateway" with "1000070" uPOKT from the account "gateway2"
    
2. :+1: 
3. :+1:

Comment (Olshansk) [reply e2e/tests/init_test.go:196]: Let's create two separate functions and use this structure: `the user stakes "application" with "1000070" uPOKT for "service" from the account "app2"`

**Review by Olshansk — CHANGES_REQUESTED** [review]

File: e2e/tests/init_test.go [hunk 3]
```diff
@@ -193,6 +193,39 @@ func (s *suite) TheUserStakesAWithUpoktFromTheAccount(actorType string, amount i
 	require.NoError(s, err, "error creating config file in %q", path.Join(os.TempDir(), configPathPattern))
 
 	configContent := fmt.Sprintf(`stake_amount: %d upokt`, amount)
+	s.Log(configContent)
```
Comment (Olshansk) [inline e2e/tests/init_test.go:196]: OPTIONAL NIT: Did you intend to keep this in or was it just for testing?

If you want to keep it going, maybe have a subtitle of what it's for?

File: e2e/tests/stake_app.feature [hunk 4]
```diff
@@ -0,0 +1,25 @@
+Feature: Stake Namespaces
+
+    Scenario: User can stake an Application
+        Given the user has the pocketd binary installed
+        And the "application" for account "app2" is not staked
+        And the account "app2" has a balance greater than "1000070" uPOKT
```
Comment (Olshansk) [inline e2e/tests/stake_app.feature:6]: Please add a comment  on where this number came from. It's one more than the default IIRRC.

File: e2e/tests/stake_app.feature [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+        When the user stakes a "application" with "1000070" uPOKT for "anvil" service from the account "app2"
+        Then the user should be able to see standard output containing "txhash:"
+        And the user should be able to see standard output containing "code: 0"
+        And the pocketd binary should exit without error
+        And the user should wait for "5" seconds
```
Comment (Olshansk) [inline e2e/tests/stake_app.feature:11]: Add a TODO to wait on an admitted staking event.

**Review by Olshansk — APPROVED** [review]
A couple small comments but nothing blocking.

PTAL at the update and :shipit:

**Pull request merged**